tiledb/sm/stats/global_stats.cc

```diff
@@ -89,8 +89,14 @@ void GlobalStats::set_enabled(bool enabled) {
 
 void GlobalStats::reset() {
   std::unique_lock<std::mutex> ul(mtx_);
-  for (auto& register_stat : registered_stats_) {
-    register_stat->reset();
+  for (auto register_stat = registered_stats_.begin(); register_stat != registered_stats_.end();) {
```

Comment:
This loop iteration is too complex to remain undocumented. It relies on the relatively little-used return value of `list::erase()` that yields the next iterator after the erased element. I'd recommend using `list::remove_if` which is far clearer in its intent.

Reply:
Not sure I follow, you think the list should be iterated multiple times,
at least once by remove_if()
and once to do ->reset()s of any remaining stats?